hazelcast/src/main/java/com/hazelcast/sql/impl/client/SqlClientResult.java

```diff
@@ -57,14 +64,26 @@ public class SqlClientResult implements SqlResult {
     /** Whether the result is closed. When {@code true}, there is no need to send the "cancel" request to the server. */
     private boolean closed;
 
+    /** Whether any SqlRow was returned from an iterator. */
+    private final AtomicBoolean returnedAnyResult = new AtomicBoolean();
```

Comment:
out of curiosity: why not simple volatile boolean?

Reply:
Leftover, will switch to volatile